lib/kitchen/directions/bake_figure.rb

```diff
@@ -3,31 +3,50 @@
 module Kitchen
   module Directions
     module BakeFigure
-      def self.v1(figure:, number:, cases: false)
+      def self.v1(figure:, number:, cases: false, custom_class: nil)
         warn 'warning! exclude unnumbered figures from `BakeFigure` loop' if figure.unnumbered?
-        figure.wrap(%(<div class="os-figure#{' has-splash' if figure.has_class?('splash')}">))
+        figure.wrap(%(<div class="os-figure#{if figure.has_class?('splash')
+                                               ' has-splash'
+                                             elsif custom_class == 'mechanism-figure'
+                                               ' has-mechanism-figure'
```

Comment:
I want to avoid `mechanism-figure` being hardcoded into this direction if possible, since it's ochem-specific. Something like

```
elsif custom_class
  " has-#{custom_class}"
end
```